src/oc_erchef/apps/chef_index/src/chef_index.erl

```diff
@@ -0,0 +1,77 @@
+%% Copyright 2015 Chef Software, Inc
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+
+-module(chef_index).
+
+-export([delete/4,
+         add/5,
+         add_async/4,
+         add_async/5
+        ]).
+
+add(TypeName, Id, DbName, IndexEjson, ReqId) ->
```

Comment:
Afaik `add_async` is only called during reindex. It follows the same code that `chef_index:add()` does, it executes the add in a `spawn` block since there isn't a specific erchef request waiting on it (`chef-server-ctl reindex` just calls the [reindex module directly](https://github.com/chef/chef-server/blob/fce335d0f84cf042ba291e2cfe7aa9204e1188d6/src/oc_erchef/priv/reindex-opc-organization#L101). We don't care about the exact time it finishes reindexing a specific item, just that it eventually does.